File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/visitor/Element.java
package jv.com.tencent.cain.design_pattern.visitor;

/**
 * @author : jiangyu
 * @date : 2020/8/22
 * @desc : xxx
 */
abstract class Element {
    protected ElementType type;

    public Element(ElementType type) {
        this.type = type;
    }

    public ElementType getType() {
        return type;
    }

    abstract void accept(Visitor visitor);
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/visitor/Element.kt
package kt.com.tencent.cain.desigin_pattern.visitor

/**
 * @author : jiangyu
 * @date   : 2020/8/23
 * @desc   : xxx
 */
abstract class Element(val elementType: VisitorElementType) {
    abstract fun accept(visitor: Visitor)
}